indexing-service/src/main/java/io/druid/indexing/common/task/HadoopIndexTask.java

```diff
@@ -149,6 +151,10 @@ public String getClasspathPrefix()
   @Override
   public TaskStatus run(TaskToolbox toolbox) throws Exception
   {
+    //Setting up UsedSegmentLister which is used by DatasourcePathSpec to get the list of used
+    //segments
+    DatasourcePathSpec.setUsedSegmentLister(new OverlordActionBasedUsedSegmentLister(toolbox));
```

Comment:
If that wouldn't help then hopefully we can manage the complexity here until we're ready to tell people "yeah, overlords are really the right way to go, and other stuff is gonna stop working"